homeassistant/components/renson/__init__.py

```diff
@@ -0,0 +1,181 @@
+
+        await hass.async_add_executor_job(renson_api.set_time, day, night)
+
+    async def set_pollution_settings(call: ServiceCall) -> None:
+        """Configure pollutions settings."""
+        day = call.data.get("day_pollution_level", "")
+        night = call.data.get("night_pollution_level", "")
+        humidity_control = call.data.get("humidity_control", "")
+        airquality_control = call.data.get("airquality_control", "")
+        co2_control = call.data.get("co2_control", "")
+        co2_threshold = call.data.get("co2_threshold", 0)
+        co2_hysteresis = call.data.get("co2_hysteresis", 0)
+
+        await renson_api.set_pollution(
+            day,
+            night,
+            humidity_control,
+            airquality_control,
+            co2_control,
+            co2_threshold,
+            co2_hysteresis,
+        )
+
+    async def set_filter_days(call: ServiceCall) -> None:
+        """Set filter dayes."""
+        days = call.data.get("days", 90)
+
+        await hass.async_add_executor_job(renson_api.set_filter_days, days)
+
+    hass.services.async_register(DOMAIN, "set_manual_level", set_manual_level)
+    hass.services.async_register(DOMAIN, "set_breeze", set_breeze)
+    hass.services.async_register(DOMAIN, "set_day_night_time", set_day_night_time)
+    hass.services.async_register(
+        DOMAIN, "set_pollution_settings", set_pollution_settings
+    )
+    hass.services.async_register(DOMAIN, "set_filter_days", set_filter_days)
+    hass.services.async_register(DOMAIN, "set_timer_level", set_timer_level)
+    hass.services.async_register(DOMAIN, "sync_time", sync_time)
+
+
+class RensonCoordinator(DataUpdateCoordinator):
+    """Data update coordinator for Renson."""
+
+    def __init__(self, hass, api, update_interval=timedelta(seconds=30)):
+        """Initialize my coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            # Name of the data. For logging purposes.
+            name="Renson",
+            # Polling interval. Will only be polled if there are subscribers.
+            update_interval=update_interval,
+        )
+        self.api = api
+
+    async def _async_update_data(self):
+        """Fetch data from API endpoint."""
+        try:
+            async with async_timeout.timeout(30):
+                return await self.hass.async_add_executor_job(self.api.get_all_data)
+        except Exception as err:
+            raise UpdateFailed(f"Error communicating with API: {err}") from err
+
+
+class RensonFirmwareCoordinator(RensonCoordinator):
+    """My custom coordinator."""
+
+    def __init__(self, hass, api):
+        """Initialize my coordinator."""
+        super().__init__(
+            hass,
+            api,
+            update_interval=timedelta(seconds=60 * 60 * 24),
+        )
+
+    async def _async_update_data(self):
+        """Fetch data from API endpoint."""
+        data = await super()._async_update_data()
```

Comment:
this is weird. This means that the firmware coordinator is also fetching all data ?

Reply:
The Renson device has only one api call that returns all data. The current firmware version is also included inside that response.